transform/mattermost-analytics/models/intermediate/product/nps/int_user_nps_score_spined.sql

```diff
@@ -0,0 +1,58 @@
+WITH first_score_day AS (
+    SELECT 
+        server_id,
+        user_id,
+        MIN(event_date) AS min_event_date
+    FROM 
+        {{ ref('int_nps_score') }}
+    GROUP BY 
+        ALL
+), spined AS (
+    SELECT 
+        date_day::date AS activity_date,
+        server_id AS server_id,
+        user_id AS user_id
+    FROM 
+        first_score_day fsd
+    LEFT JOIN 
+        {{ ref('telemetry_days') }} td 
+    ON 
+        date_day::date >= min_event_date
+), score_cte AS (
+    SELECT 
+        sp.activity_date,
+        sp.server_id,
+        sp.user_id,
+        nps_score.score,
+        COALESCE(nps_score.user_role, FIRST_VALUE(nps_score.user_role IGNORE NULLS) OVER (
```

Comment:
Another option would be to pivot user role so that there's a protractors/detractors/passive column per each role type. This would reduce the number of rows in the final table and make aggregations easier. Also given that snowflake is columnar, the queries would be more efficient (it would eliminate the need for filtering each row).

Reply:
Sounds like a great idea, I'm all for it